### PR TITLE
mcs: support alloc ID for scheduling server

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

We need to allocate ID for building the operator in the scheduling server. But the ID is maintained by the API server, so we have to call RPC to get the ID.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/scheduling/server/server.go [hunk 1]
```diff
@@ -165,8 +169,53 @@ func (s *Server) Run() error {
 
 func (s *Server) startServerLoop() {
 	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(s.ctx)
-	s.serverLoopWg.Add(1)
+	s.serverLoopWg.Add(2)
 	go s.primaryElectionLoop()
+	go s.updateMemberLoop()
+}
+
+func (s *Server) updateMemberLoop() {
```
Comment (JmPotato) [inline pkg/mcs/scheduling/server/server.go:177]: What about naming it `updateAPIServerMember`?

File: tests/testutil.go [hunk 2]
```diff
@@ -33,8 +33,23 @@ import (
 
 var once sync.Once
 
-// InitLogger initializes the logger for test.
-func InitLogger(cfg *tso.Config) (err error) {
+// InitSchedulingLogger initializes the logger for test.
+func InitSchedulingLogger(cfg *sc.Config) (err error) {
```
Comment (lhy1024) [inline tests/testutil.go:37]: Does resource manager server need similar function?

Comment (rleungx) [reply tests/testutil.go:37]: We need further refactoring to merge them. For now, we don't use it.

**Review by lhy1024 — APPROVED** [review]

File: pkg/mcs/scheduling/server/cluster.go [hunk 3]
```diff
@@ -128,11 +136,29 @@ func (c *Cluster) GetSchedulerConfig() sc.SchedulerConfigProvider { return c.per
...
 func (c *Cluster) GetStoreConfig() sc.StoreConfigProvider { return c.persistConfig }
 
+// AllocID allocates a new ID.
+func (c *Cluster) AllocID() (uint64, error) {
+	cli := c.apiServerLeader.Load().(pdpb.PDClient)
+	if cli == nil {
+		c.checkMembershipCh <- struct{}{}
+		return 0, errors.New("API server leader is not found")
+	}
+	resp, err := cli.AllocID(context.Background(), &pdpb.AllocIDRequest{Header: &pdpb.RequestHeader{ClusterId: c.clusterID}})
```
Comment (JmPotato) [inline pkg/mcs/scheduling/server/cluster.go:146]: What about using a passed-in context by calling `AllocID(ctx)`?

**Review by JmPotato — COMMENTED** [review]
The rest are LGTM.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**